lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -409,6 +409,41 @@ bool SinkCode::run(Function *F, const CompilationContext &cctx) {
   // For each node:
   for (auto &N : nodes) {
     auto *node = &N;
+    // Sink Reshape below BatchNormalization.
+    if (auto *BN = dyn_cast<BatchNormalizationNode>(node)) {
+      if (auto *RS = dyn_cast<ReshapeNode>(BN->getInput())) {
+        auto inDims = RS->getInput().dims();
+        auto outDims = RS->getType(0)->dims();
```

Comment:
We already know what RS is. can just getResult and type from there.

Reply:
@ayermolo, I am not sure I understood you point. Could you please elaborate?